lock-impl/src/test/java/com/palantir/lock/impl/RateLimitingLockServiceTest.java

```diff
@@ -0,0 +1,158 @@
+/**
+ * Copyright 2017 Palantir Technologies
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.palantir.lock.impl;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import com.google.common.collect.ImmutableSortedMap;
+import com.palantir.lock.LockMode;
+import com.palantir.lock.LockRefreshToken;
+import com.palantir.lock.LockRequest;
+import com.palantir.lock.LockService;
+import com.palantir.lock.StringLockDescriptor;
+
+public class RateLimitingLockServiceTest {
+    private static final LockRequest MOCK = LockRequest.builder(ImmutableSortedMap.of(StringLockDescriptor.of("mock"), LockMode.WRITE))
+            .build();
+    public static final int AVAILABLE_THREADS = 100;
+    public static final int NUM_CLIENTS = 10;
+    public static final int LOCAL_TC_LIMIT = AVAILABLE_THREADS / 2 / NUM_CLIENTS;
```

Comment:
`x / y / z` is well-defined in Java but not mathematically. Could you add brackets to make it clear? (also in the above class)